randovania/games/prime_hunters/layout/prime_hunters_teleporters.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import TYPE_CHECKING
+
+from randovania.game_description import default_database
+from randovania.game_description.db.dock_node import DockNode
+from randovania.layout.lib.teleporters import TeleporterConfiguration, TeleporterShuffleMode
+
+if TYPE_CHECKING:
+    from randovania.game_description.db.node_identifier import NodeIdentifier
+
+
+@dataclasses.dataclass(frozen=True)
+class HuntersTeleporterConfiguration(TeleporterConfiguration):
+    @property
+    def valid_targets(self) -> list[NodeIdentifier]:
+        if self.mode in {
+            TeleporterShuffleMode.ONE_WAY_TELEPORTER,
+            TeleporterShuffleMode.ONE_WAY_TELEPORTER_REPLACEMENT,
+        }:
+            game_description = default_database.game_description_for(self.game)
```

Comment:
This is going to be the 4th copy paste of this function. This needs to be refactored into some common module.

Reply:
PR has been updated with the latest refactor changes.